lib/operations.coffee

```diff
@@ -108,6 +108,15 @@ exports.execute = (image, operations, options) ->
 
 	emitter = new EventEmitter()
 
+	# There is an edge case where the event emitter instance
+	# emits the `end` event before the client is able to
+	# register a listener for it.
+	emitterOn = emitter.on
```

Comment:
@emirotin I'm saving a copy of the original function since I'm overwriting it below. It'll cause infinite recursion otherwise.